cmd/prometheus/main.go

```diff
@@ -212,6 +212,9 @@ func main() {
 	a.Flag("storage.remote.read-concurrent-limit", "Maximum number of concurrent remote read calls. 0 means no limit.").
 		Default("10").IntVar(&cfg.web.RemoteReadConcurrencyLimit)
 
+	a.Flag("storage.remote.read-max-chunks-in-frame", "Maximum number of chunks in single frame for STREAMED_XOR_CHUNKS remote read response type. Each chunk corresponds roughly to (~3B * 120 samples) + 32B. Default is 1000 which is roughly (1000 * ~400B) + labelset, so approx. 0.4MB per frame .").
+		Default("1000").IntVar(&cfg.web.RemoteReadMaxChunksInFrame)
+
```

Comment:
What happens when a single series has more than chunks than this?

Reply:
I actually don't see where this is enforced...

Reply:
wait. I think there is misunderstanding. This is not a limit. It's essentially control on how much you want to buffer / how large frames are.

You can have many frames per series. How we can make it more clear here?

Reply:
How about specifying this in bytes, so it works across future types?

Reply:
@bwplotka yup, I misunderstood.  I see how this works now.  Thanks.

Reply:
Should we do it by bytes then? `storage.remote.read-max-bytes-in-frame` 

I think we might be able to calculate on the stream side, but it might be bit brittle and complex to implement.

Reply:
We'd only need something that's good enough. The idea is more to stop massive frames, than to get it within the byte.

Reply:
I still think we should either do this in bytes, or just leave it hardcoded.